tools/cold-backup/run.sh

```diff
@@ -0,0 +1,434 @@
+        fi
+        local kafka_ss="$(oc get statefulset $cluster_name-kafka -o name --ignore-not-found)"
+        if [[ -n $kafka_ss ]]; then
+            kubectl scale $kafka_ss --replicas 0
+            __wait_for pod delete strimzi.io/name=$cluster_name-kafka
+        fi
+        local zoo_ss="$(oc get statefulset $cluster_name-zookeeper -o name --ignore-not-found)"
+        if [[ -n $zoo_ss ]]; then
+            kubectl scale $zoo_ss --replicas 0
+            __wait_for pod delete strimzi.io/name=$cluster_name-zookeeper
+        fi
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__export_env() {
+    local op_pod="$(kubectl get pods | grep strimzi-cluster-operator | grep Running | cut -d " " -f1)"
+    kubectl exec -i $op_pod -- env | grep "VERSION" | sed -e "s/^/declare -- /" > "$__TMP/env"
+    local filter="data-$CLUSTER_NAME-zookeeper"
+    ZOO_REPLICAS=$(kubectl get kafka $CLUSTER_NAME -o yaml | yq eval ".spec.zookeeper.replicas" -)
+    ZOO_PVC_SIZE=$(kubectl get pvc $filter-0 -o yaml | yq eval ".spec.resources.requests.storage" -)
+    ZOO_PVC_CLASS=$(kubectl get pvc $filter-0 -o yaml | yq eval ".spec.storageClassName" -)
+    JBOD_VOL_NUM=$(kubectl get pvc | grep $CLUSTER_NAME-kafka-0 | wc -l)
+    if ((JBOD_VOL_NUM > 1)); then
+        filter="data-0-$CLUSTER_NAME-kafka"
+    else
+        filter="data-$CLUSTER_NAME-kafka"
+    fi
+    KAFKA_REPLICAS=$(kubectl get kafka $CLUSTER_NAME -o yaml | yq eval ".spec.kafka.replicas" -)
+    KAFKA_PVC_SIZE=$(kubectl get pvc $filter-0 -o yaml | yq eval ".spec.resources.requests.storage" -)
+    KAFKA_PVC_CLASS=$(kubectl get pvc $filter-0 -o yaml | yq eval ".spec.storageClassName" -)
+    declare -px ZOO_REPLICAS ZOO_PVC_SIZE ZOO_PVC_CLASS JBOD_VOL_NUM \
+        KAFKA_REPLICAS KAFKA_PVC_SIZE KAFKA_PVC_CLASS >> "$__TMP/env"
+}
+
+__export_res() {
+    local name="$1"
+    local label="${2-}"
+    if [[ -n $name ]]; then
+        local crs=$(kubectl get $name -o name -l "$label" --ignore-not-found)
+        if [[ -n $crs ]]; then
+            echo "Exporting $name"
+            # delete runtime metadata expression
+            local exp="del(.metadata.namespace, .items[].metadata.namespace, \
+                .metadata.resourceVersion, .items[].metadata.resourceVersion, \
+                .metadata.selfLink, .items[].metadata.selfLink, \
+                .metadata.uid, .items[].metadata.uid, \
+                .status, .items[].status)"
+            local id=$(printf $name | sed 's/\//-/g;s/ //g')
+            kubectl get $crs -o yaml | yq eval "$exp" - > $__TMP/resources/$NAMESPACE-$id.yaml
+        fi
+    fi
+}
+
+__rsync() {
+    local source="$1"
+    local target="$2"
+    if [[ -n $source && -n $target ]]; then
+        echo "Rsync from $source to $target"
+        local pod_name="strimzi-backup"
+        local patch=$(sed "s/\$pvc/$pvc/g" $__HOME/patch.json)
+        kubectl run $pod_name --image="dummy" --restart="Never" --overrides="$patch"
```

Comment:
What is the purpose of the dummy image here? Our tests are designed that CO will always try to download that image. You have to somehow modify `imagepullpolicy`...

Reply:
Not sure what you mean. That image is needed to spin up a maintenance pod that is at the heart of the script.